lib/posix/pthread_barrier.c

```diff
@@ -160,6 +160,40 @@ int pthread_barrier_destroy(pthread_barrier_t *b)
 	return 0;
 }
 
+int pthread_barrierattr_init(pthread_barrierattr_t *attr)
+{
+	__ASSERT_NO_MSG(attr != NULL);
+
+	((pthread_barrierattr_t *) attr)->pshared = PTHREAD_PROCESS_DEFAULT;
+
+	return 0;
+}
+
+int pthread_barrierattr_setpshared(pthread_barrierattr_t *attr, int pshared)
+{
+	if ((pshared != PTHREAD_PROCESS_PRIVATE) || (pshared != PTHREAD_PROCESS_PUBLIC)) {
```

Comment:
Logic was slightly incorrect
```suggestion
	if (pshared != PTHREAD_PROCESS_PRIVATE && pshared != PTHREAD_PROCESS_PUBLIC) {
```